src/query/functions/src/scalars/datetime.rs

```diff
@@ -131,6 +136,125 @@ fn int64_domain_to_timestamp_domain<T: AsPrimitive<i64>>(
     })
 }
 
+fn register_convert_timezone(registry: &mut FunctionRegistry) {
+    // 2 arguments function [target_timezone, src_timestamp]
+    registry.register_passthrough_nullable_2_arg::<StringType, StringType, TimestampType, _, _>(
+        "convert_timezone",
+        |_, _, _| FunctionDomain::MayThrow,
+        eval_convert_timezone,
+    );
+
+    // 3 arguments function [target_timezone, src_timestamp, src_timezone]
```

Comment:
For compatibility with snowflake, the order of the arguments should be `src_timezone`, `target_timezone`, `src_timestamp`
https://docs.snowflake.com/en/sql-reference/functions/convert_timezone